modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

```diff
@@ -0,0 +1,77 @@
+:_content-type: PROCEDURE
+
+[id="adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance"]
+= Adding or removing pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance
+
+You can add or remove your choice of pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance that is cached by {prod-short} to support air-gapped, offline, and proxy-restricted environments.
+
+Completing this procedure results in an optional, custom build of the embedded Open VSX registry for use in workspaces of your organization’s Che.
+
+.Procedure
+
+include::example$snip_{project-context}-steps-to-add-or-remove-extensions-in-openvsx-sync-json.adoc[]
+
+** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add it by including an object with the extension `id` in the format __<published_by>__.__<unique_identifier>__ that you can collect from the extension page on link:https://open-vsx.org[open-vsx.org]:
++
+[source,json,subs="+quotes"]
+----
+    {
+        "id": "__<published_by>__.__<unique_identifier>__"
+    }
+----
++
+[TIP]
+====
+The latest extension version on link:https://open-vsx.org[open-vsx.org] is the default. Alternatively, you can add `"version": "__<extension_version>__"` on a new line to specify a version.
+====
+
+** If it is an extension from link:https://marketplace.visualstudio.com/VSCode[Visual Studio Marketplace] that is not available on link:https://open-vsx.org[Open VSX] yet, you can ask the extension author to publish it on link:https://open-vsx.org[open-vsx.org] according to the link:https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension[instructions] so that you can add it as an extension that is published on link:https://open-vsx.org[open-vsx.org].
```

Comment:
```suggestion
** If an extension is only available from link:https://marketplace.visualstudio.com/VSCode[Visual Studio Marketplace], but not link:https://open-vsx.org[Open VSX], you can ask the extension author to publish it on link:https://open-vsx.org[open-vsx.org] according to the link:https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension[instructions].

 Otherwise the link:https://aka.ms/vsmarketplace-ToU[Terms Of Use] for the link:https://marketplace.visualstudio.com/VSCode[Visual Studio Marketplace] allow using extensions only in Microsoft's products and services.
```

Reply:
@nickboldt, can I append a phrase at the end of the sentence: " if the extension's publisher is not Microsoft."?

Reply:
Doesn't matter who the publisher is. It matters that the TOU for the MS registry says "only with In-Scope Products and Services". 

"Marketplace Offerings are intended for use only with In-Scope Products and Services and you may install and use Marketplace Offerings only with In-Scope Products and Services."

The list of products is in paragraph one: "Microsoft Visual Studio, Visual Studio for Mac,
Visual Studio Code, GitHub Codespaces, Azure DevOps, Azure DevOps Server, and successor products
and services (the “In-Scope Products and Services”) offered by us and GitHub, Inc. (“GitHub”)". 

So... no. Just say "don't use them" and link to the TOU.

Reply:
@nickboldt, what I meant is we can't seriously suggest that our customers contact Microsoft with requests to have Microsoft's extensions published on Open VSX website.

``` 
If an extension is only available from link:https://marketplace.visualstudio.com/VSCode[Visual Studio Marketplace], but not link:https://open-vsx.org[Open VSX], you can ask the extension author to publish it on link:https://open-vsx.org[open-vsx.org] according to the link:https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension[instructions].
```

Reply:
> ask the extension author

the author != Microsoft in most cases.

Reply:
@nickboldt, yes, exactly. This is what I'm saying too.

Reply:
@nickboldt, can I add `third-party` to it? 
```
** If an extension is only available from ... Visual Studio Marketplace
```
`>`
```
** If a third-party extension is only available from ... Visual Studio Marketplace
```

Reply:
@nickboldt, and I'm also thinking of one more bullet list item:

> If there is no Open VSX equivalent for an extension that you use and that is published by Microsoft on ... Visual Studio Marketplace, consider informing the VSX team by opening an issue at https://github.com/open-vsx/publish-extensions/issues.

Something in that vein... WDYT?

Reply:
@nickboldt, OMG:

https://marketplace.visualstudio.com/items?itemName=ms-python.python > https://github.com/Microsoft/vscode-python

![image](https://user-images.githubusercontent.com/78564304/208753272-94c4aa1c-ae4c-46fa-83a7-da342b232ead.png)

Another reason to add my suggestion in my previous comment above ^.

Reply:
And Microsoft's original extensions have an MIT license: https://github.com/microsoft/vscode-cpptools/blob/main/LICENSE.md

Reply:
Kinda getting into the weeds here. 

If you want a tip in both up/downstream that says something like you suggested above, I'm good with that:

```
If there is no Open VSX equivalent of an extension that you use consider informing the Open VSX team at https://github.com/open-vsx/publish-extensions/issues to let them know.
```

Of course I don't know if the Open VSX maintainers will just rebuild/republish/fork random extensions just because you ask for it... so this might be a fool's errand.